### PR TITLE
docs(coc): introduce CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by gkatsev**
~~@videojs/core-committers please take a look. I used [weallbehave](https://github.com/WeAllJS/weallbehave) to generate the CoC.~~
It is now using [Contributor Covenant](http://contributor-covenant.org). See [this comment](https://github.com/videojs/video.js/pull/4160#issuecomment-286178821)

### CODE REVIEW

**Comment by misteroneill** [comment]
I'd say most of this is uncontroversial, but...

> Any attempt to present "reverse-ism" versions of the above as violations. Examples of reverse-isms are "reverse racism", "reverse sexism", "heterophobia", and "cisphobia".

This seems to be explicitly endorsing a specific worldview and form of identity politics that I don't agree with. I think it directly contradicts the "Pledge" to refrain from discrimination. Everyone should receive equal treatment under any regime of rule or law.

**Comment by zkat** [comment]
@misteroneill The purpose of acknowledging the "reverse-ism" thing is a matter of identifying where and how oppression happens in society. Heterosexual people, as people, can face oppression, but they are not oppressed in society _for being heterosexual_, because their heterosexuality is part of the accepted and preferred norm of the existing societal power structures. Same goes for being white, and these things go far beyond opinion: they have research and centuries of evidence backing them up.

This clause, by the way, does not preclude you from asking people to refrain from unwarranted hostility. I still don't think it's ok for people to arbitrarily attack anyone based on anything, but if a conflict happens because someone said something against a person of color, and the person of color responds with "white people are the worst", that's a situation where the person of color would not be considered to be in violation of this. It's still ok to expect people to be kind and respectful, as that benefits everyone (the WeAllJS CoC is heavily about reducing conflict temperatures -- aggressive language is still no good. It's just not considered racist/sexist/etc).

**Comment by misteroneill** [comment]
@zkat No worries, your feedback and clarification is appreciated.

> ...if a conflict happens because someone said something against a person of color, and the person of color responds with "white people are the worst", that's a situation where the person of color would not be considered to be in violation of this.

That's exactly how I interpreted the clause and it's an attitude I, personally, don't want to endorse.

I readily accept that being white, for example, comes with privilege that insulates one from systemic racism. However, I disagree that it is acceptable to express prejudice, bias, or hatred against someone because of their accidental membership in a privileged group.

I worry that this clause may preclude people from contributing to Video.js. Obviously, I don't speak for the project maintainers as a whole - only myself - but I've said my peace.

**Comment by OwenEdwards** [comment]
My two cents: I see what @misteroneill is saying, and I'm wondering if the intent of the CoC is more:

> Any attempt the deflect unacceptable behavior by using "reverse-isms", or equating "reverse-isms" with oppressive speech about a minority group (which can be an attempt to minimize the legacy and impact of long-standing and systemic oppression of minorities).

**Comment by gkatsev** [comment]
I've updated this PR to use the [Contributor Covenant](http://contributor-covenant.org). I did copy over the "Other Community Standards" and "Further Enforcement" sections from weallbehave, which I attributed in the last section.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Contributor Covenant Code of Conduct
+
+Please note that this project is released with a Contributor Code of Conduct. By participating in this project you agree to abide by its terms.
+
```
Comment (BrandonOCasey) [inline CODE_OF_CONDUCT.md]: Do we want a Table of Contents here? Do we want to run the remark linter/fixer on this file?

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+Additionally, as a project hosted on npm, is is covered by [npm, Inc's Code of Conduct](https://www.npmjs.com/policies/conduct).
+
+Enforcement of those guidelines after violations overlapping with the above are the responsibility of the entities, and enforcement may happen in any or all of the services/communities.
+
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at gary@videojs.com. All
```
Comment (BrandonOCasey) [inline CODE_OF_CONDUCT.md]: I wonder if we should mention that they can `@videojs/contributors` or email. I also think that emailing gary@videojs.com is a bit weird, but I don't think I have a quicker/better solution.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (gkatsev) [inline CODE_OF_CONDUCT.md]: Yeah, we might as well. Though, the more we change it, the harder it'll be to update if the contributor covenant updates again.

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (gkatsev) [inline CODE_OF_CONDUCT.md]: core-contributors if anything, but enforcement of the CoC is tricky and I don't want to push it on people who don't want to do it.

Comment (gkatsev) [inline CODE_OF_CONDUCT.md]: I can also add that they can message me on slack.

**Comment by mister-ben** [comment]
I think this version is much better. It covers all the bases without getting bogged down in specifics that while well intentioned may be misinterpreted.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (gkatsev) [inline CODE_OF_CONDUCT.md]: Updated

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+Maintainers and other leaders who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project's leadership. These may include anything from removal from the maintainer team to a permanent ban from the community.
+
+Additionally, as a project hosted on both GitHub and npm, [their own Codes of Conducts may be applied against maintainers of this project](#other-community-standards), externally of this project's procedures.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
+available at [http://contributor-covenant.org/version/1/4][version]
+
+The [Other Community Standards](#other-community-standard) and [Enforcement][#enforcement] sections
```
Comment (OwenEdwards) [inline CODE_OF_CONDUCT.md]: Should this be `[Enforcement](#enforcement)` instead?

Comment (gkatsev) [inline CODE_OF_CONDUCT.md]: Yes

**Comment by gkatsev** [comment]
@videojs/core-committers please take a look. Thanks.

**Comment by hartman** [comment]
👍 thx !

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+As a project on GitHub, this project is additionally covered by the [GitHub Community Guidelines](https://help.github.com/articles/github-community-guidelines/).
+
+Additionally, as a project hosted on npm, is is covered by [npm, Inc's Code of Conduct](https://www.npmjs.com/policies/conduct).
+
+Enforcement of those guidelines after violations overlapping with the above are the responsibility of the entities, and enforcement may happen in any or all of the services/communities.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at gary@videojs.com or @gkatsev on [slack][]. All
```
Comment (dmlap) [inline CODE_OF_CONDUCT.md]: Is that slack link complete?

Comment (gkatsev) [inline CODE_OF_CONDUCT.md]: Yup. It's shorthand for `[slack][slack]`, so, it looks below for the url.

**Pull request merged**